commit 7ab53c3a8e510035b29574aa88abe0d3397b55f2
Author: Wolfgang Rohdewald <wolfgang@rohdewald.de>
Date:   Wed Sep 10 10:17:52 2014 +0200

    Ticket 7667: twisted.spread: Assume that Python always has unicode support enabled
    
    Many other places in the assume this too, and
    zope.interface also does (see interface/_compat.py)
    
    Python 2.5 can still be compiled without unicode support,
    Python 2.6 cannot
    Python 2.7 can, see http://bugs.python.org/issue8767 (fix not applied to 2.6)

diff --git twisted/spread/jelly.py twisted/spread/jelly.py
index df1956d..dfff192 100644
--- twisted/spread/jelly.py
+++ twisted/spread/jelly.py
@@ -62,7 +62,7 @@ The same rule applies for C{frozenset} and C{sets.ImmutableSet}.
 @author: Glyph Lefkowitz
 """
 
-from twisted.python.compat import _PY3
+from twisted.python.compat import _PY3, unicode
 
 # System Imports
 import pickle
@@ -71,7 +71,6 @@ import warnings
 import decimal
 from functools import reduce
 from types import StringType
-from types import UnicodeType
 from types import IntType
 from types import TupleType
 from types import ListType
@@ -488,7 +487,7 @@ class _Jellier:
                         self.jelly(obj.im_self),
                         self.jelly(obj.im_class)]
 
-            elif UnicodeType and objType is UnicodeType:
+            elif objType is unicode:
                 return ['unicode', obj.encode('UTF-8')]
             elif objType is NoneType:
                 return ['None']
@@ -692,10 +691,7 @@ class _Unjellier:
 
 
     def _unjelly_unicode(self, exp):
-        if UnicodeType:
-            return unicode(exp[0], "UTF-8")
-        else:
-            return Unpersistable("Could not unpersist unicode: %s" % (exp[0],))
+        return unicode(exp[0], "UTF-8")
 
 
     def _unjelly_decimal(self, exp):
@@ -1038,8 +1034,7 @@ class SecurityOptions:
                              "date": 1,
                              "timedelta": 1,
                              "NoneType": 1}
-        if hasattr(types, 'UnicodeType'):
-            self.allowedTypes['unicode'] = 1
+        self.allowedTypes['unicode'] = 1
         self.allowedTypes['decimal'] = 1
         self.allowedTypes['set'] = 1
         self.allowedTypes['frozenset'] = 1
diff --git twisted/topfiles/7667.removal twisted/topfiles/7667.removal
new file mode 100644
index 0000000..7b80a13
--- /dev/null
+++ twisted/topfiles/7667.removal
@@ -0,0 +1 @@
+twisted.spread.jelly: Support for Python 2 compiled without unicode support has been removed.
